syntax = "proto3";

package masterworker;

message Master_to_Worker {
	repeated MasterQuery masterquery = 1;
}

message MasterQuery{
	string file_path = 1;
	int64 file_offset = 2;
	bool map_reduce = 3; // true: map, false: reduce
	int64 data_size = 4;
	int64 ID_assigned_to_worker = 5;
}

message Worker_to_Master {
	string file_path = 1;
}